File: 3172-divisible-and-non-divisible-sums-difference/3172-divisible-and-non-divisible-sums-difference.ts
function differenceOfSums(n: number, m: number): number {
  let sum: number = n*(n+1)/2
    let sum1: number = 0

    for (let counter: number = 0; counter <= n; counter +=m){
        sum1 += counter
    }
    return sum - sum1 - sum1
};

File: 3172-divisible-and-non-divisible-sums-difference/3172-divisible-and-non-divisible-sums-difference.java
class Solution {
    public int differenceOfSums(int n, int m) {
        
        int sum = (n*(n+1))/2;
        int sum1 = 0;

        for (int counter = 0; counter <= n; counter += m){
            sum1 += counter;
        }
        return sum - sum1*2;
    }
}